src/ess/livedata/dashboard/widgets/correlation_histogram_widget.py

```diff
@@ -19,83 +18,49 @@ def __init__(
-            disabled=True,
-            button_type='primary',
+        self._config_2d_button = pn.widgets.Button(
+            name='2D Correlation Histogram', disabled=True, button_type='primary'
         )
-        self._config_button.on_click(self._on_config_button_click)
+        self._config_2d_button.on_click(self._on_config_2d_button_click)
 
-        # Register for updates and set up selection watching
+        # Register for updates
         self._correlation_histogram_controller.register_update_subscriber(
-            self._update_timeseries_list
+            self._update_button_states
         )
-        self._tabulator.param.watch(self._on_selection_change, 'selection')
 
         # Initial update
-        self._update_timeseries_list()
+        self._update_button_states()
 
         # Container for modals - they need to be part of the served structure
         self._modal_container = pn.Column()
 
-    def _update_timeseries_list(self) -> None:
-        """Update the tabulator with current timeseries data."""
+    def _update_button_states(self) -> None:
+        """Update button states based on available timeseries."""
         timeseries_keys = self._correlation_histogram_controller.get_timeseries()
-        data = pd.DataFrame(
-            [
-                {
-                    'source_name': key.job_id.source_name,
-                    'job_id': str(key.job_id.job_number)[:8],
-                    'output_name': key.output_name or '',
-                    'result_key': key,
-                }
-                for key in timeseries_keys
-            ]
-        )
-        if timeseries_keys:
-            self._result_key = data.pop('result_key')
-        self._tabulator.value = data
-        self._on_selection_change()  # Update button state
-
-    def _on_selection_change(self, *args: Any) -> None:
-        """Handle selection changes in the tabulator."""
-        selection = self._tabulator.selection
-        num_selected = len(selection)
-
-        # Enable button only for 1 or 2 selections
-        self._config_button.disabled = num_selected not in (1, 2)
-
-        # Update button text based on selection
-        if num_selected == 1:
-            self._config_button.name = 'Create 1D Configuration'
-        elif num_selected == 2:
-            self._config_button.name = 'Create 2D Configuration'
-        else:
-            self._config_button.name = 'Create Configuration'
-
-    def _on_config_button_click(self, event: Any) -> None:
-        """Handle configuration button clicks."""
-        selection = self._tabulator.selection
-        indices = self._tabulator.value.iloc[selection].index
-        config = self._correlation_histogram_controller.create_config(
-            [self._result_key.iloc[i] for i in indices]
-        )
+        num_timeseries = len(timeseries_keys)
+        self._config_1d_button.disabled = num_timeseries < 1
+        self._config_2d_button.disabled = num_timeseries < 2
```

Comment:
Do these work when more than 1 or 2 time series are selected, respectively? The code in `correlation_histogram.py` looks like it uses the first 1 or first 2 coords. This could be confusing, can you change it to require exactly 1 or 2 coords?

Reply:
This code is about enabling the buttons: If there are 5 timeseries then both plots are possible. The user can then choose between those in 1 or 2 dropdown menus.